bowling/directoryContent.cpp

```diff
@@ -0,0 +1,59 @@
+#include "directoryContent.hpp"
+#include <algorithm>
+#include <fstream>
+namespace fs = std::filesystem;
+
+directoryContent::directoryContent(char dir[]) {
+  fs::path directory(dir);
+  directory_ = directory;
+  searchFilesInDirectory();
+  sortFilesInVector();
+  readFiles();
+};
+
+void directoryContent::searchFilesInDirectory() {
+  if (fs::exists(directory_) && fs::is_directory(directory_)) {
+    for (auto const &entry : fs::directory_iterator(directory_)) {
+      auto pathName = entry.path();
+      if (fs::is_regular_file(entry.status()))
+        filesInDirectory.emplace_back(entry.path());
```

Comment:
Could this be replaced with std::for_each using https://en.cppreference.com/w/cpp/filesystem/directory_iterator/begin ?